qemu/tests/usb_smartcard_sharing.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import re
+import os
+
+from virttest import error_context
+from virttest import env_process
+from virttest import data_dir
+from virttest import utils_misc
+from virttest.qemu_devices import qdevices
+from virttest.utils_params import Params
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test usb smartcard sharing:
+    Guest  - user of the smartcard
+    Client - owner of the smartcard (use host as client in this case)
+
+    1) Preprocess VM
+    2) Check client configurations
+    4) Start smartcard sharing via spice
+    5) Check guest configurations
+    6) Check the smartcard in guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _client_config_check():
+        status = True
+        err_msg = ''
+        gui_group = "Server with GUI"
+        out = process.getoutput('yum group list --installed',
+                                allow_output_check='stdout', shell=True)
+        obj = re.search(r"(Installed Environment Groups:.*?)^\S",
+                        out, re.S | re.M)
+        if not obj or gui_group not in obj.group(1):
+            gui_groupinstall_cmd = "yum groupinstall -y '%s'" % gui_group
+            s, o = process.getstatusoutput(gui_groupinstall_cmd, shell=True)
+            if s:
+                status = False
+                err_msg = "Fail to install '%s' on client, " % gui_group
+                err_msg += "output: %s" % o
+                return (status, err_msg)
+
+        virt_viewer_cmd = "rpm -q virt-viewer || yum install -y virt-viewer"
+        s, o = process.getstatusoutput(virt_viewer_cmd, shell=True)
+        if s:
+            status = False
+            err_msg = "Fail to install 'virt-viewer' on client, "
+            err_msg += "output: %s" % o
+            return (status, err_msg)
+
+        # unpack fake-smartcard database
+        sc_db = params.get('sc_db_tar', 'fake-smartcard.tar.gz')
+        sc_db_src = os.path.join(data_dir.get_deps_dir(), 'smartcard', sc_db)
```

Comment:
In this file,  `data_dir.get_deps_dir()` without any parameter (`target`),  will get `tp-qemu/qemu/deps` , but the file you upload is `tp-qemu/deps/smartcard*****`, i am curious why the test passed in your side.

The correct usage is `data_dir.get_deps_dir('smartcard')`, which will check the parent directory.
And suggest saving `fake-smartcard.tar.gz` in `to-qemu/qemu/deps` as it is case specific, not for the module under provider dir.

Reply:
Updated, thanks!